Sources/Widgets/Widgets3D/SplineWidget/behavior.js

```diff
@@ -273,12 +273,12 @@ export default function widgetBehavior(publicAPI, model) {
       return macro.VOID;
     }
 
-    model.manipulator.setOrigin(model.activeState.getOrigin());
     model.manipulator.setNormal(model.camera.getDirectionOfProjection());
     const worldCoords = model.manipulator.handleEvent(
       callData,
       model.openGLRenderWindow
     );
+    model.manipulator.setOrigin(worldCoords);
```

Comment:
it does not seem to make sense to `handle event` with an out of date origin

Reply:
To me it is the updated value that we  are using here, set in the update function of the example. @laurennlam How would you explain it?

Reply:
In the example, when the slice is moved, then the origin of the manipulator is updated with the slice center.
Then, with this new origin, we can compute/project the correct point1 and point 2.
It couldn't work before because we used the point1 of the current state. But this point was never correctly set.

Reply:
@floryst can you refresh me on the origin of the manipulator vs origin of the points? I don't quite understand this change. See here: https://github.com/Kitware/vtk-js/pull/1717/files#diff-09c501501761db156d6f7d2d7b93d812c1fccc886aac5a921dbda49b32481362L387-R388

Reply:
Origin of the plane manipulator sets the plane origin. Origin of the points are the coordinates of each point.

Just as a sanity check, is the plane manipulator's origin correct prior to calling [manipulator.handleEvent here](https://github.com/Kitware/vtk-js/pull/1717/files#diff-09c501501761db156d6f7d2d7b93d812c1fccc886aac5a921dbda49b32481362R384)?

In addition, should we be updating the manipulator like this in behavior? [Recent discussions](https://github.com/Kitware/vtk-js/issues/1718#issuecomment-773171223) on this should be considered. If this widget is expected to only ever operate on a plane orthogonal to the view direction, then keeping it is fine.

Reply:
To answer your question @floryst the manipulator's origin is correct before the 1st call to manipulator.handleEvent.
@finetjul is there a use case which is not using orthogonal planes?

Reply:
> @finetjul is there a use case which is painting/placing shapeWidgets on non orthogonal planes?

Yes. That was the goal of #1422.